File: crates/taflow-core/src/stream/fractal_dimension_test.rs
use super::fractal_dimension::FractalDimension;

#[test]
fn warmup_and_reset_are_consistent() {
    let mut state = FractalDimension::new(3).unwrap();
    assert_eq!(state.append(1.0), None);
    assert_eq!(state.append(2.0), None);
    assert!(state.append(3.0).is_some());
    state.reset();
    assert_eq!(state.value(), None);
}

File: tests/fractal_dimension_test.py
import numpy as np

from taflow import FractalDimension


def test_fractal_dimension_lifecycle_and_reset():
    state = FractalDimension(timeperiod=4)
    state.extend([1.0, 2.0, 3.0, 4.0])
    assert np.isfinite(state.compute()[-1])
    assert len(state) == 4
    state.reset()
    assert state.value is None
